### DISCUSSION TITLE
What's the best way to replace when + includes

### DISCUSSION THREAD

**Discussion by syabro** [gvergnaud/ts-pattern#263] [Q&A]
In my code

```ts
const COMPONENTS_EXTENSIONS = ['.vst', '.vst3', '.component', '.au', '.aaxplugin', '.app']
      
match(extension)
  .with('pkg', async () => do1())
  .when((_ext) => COMPONENTS_EXTENSIONS.includes(_ext), async () => do2())
  .otherwise(() => {
    throw new Error(`Unsupported file type: ${file}`)
  })
```

**Comment by gvergnaud** [comment]
You can use P.union, or pass these values directly to with:

```ts
match(extension)
  .with(
    P.union(".vst", ".vst3", ".component", ".au", ".aaxplugin", ".app"),
    async () => do2()
  )
  // OR
  .with(".vst", ".vst3", ".component", ".au", ".aaxplugin", ".app", async () =>
    do2()
  )

```

**Reply by syabro** [reply]
Emmm, array constant like `COMPONENTS_EXTENSIONS` must be used. 
Passing values manually sure works
I tried `... COMPONENTS_EXTENSIONS` but gets ts errors

**Reply by gvergnaud** [reply]
It will work if you declare your `COMPONENTS_EXTENSIONS` variable as const

```ts
import {match} from 'ts-pattern'

const COMPONENTS_EXTENSIONS = ['.vst', '.vst3', '.component', '.au', '.aaxplugin', '.app'] as const
      
declare const extension: string

match(extension)
  .with('pkg', async () => {})
  .with(...COMPONENTS_EXTENSIONS, async (x) => {})
   //                                    ^?
  .otherwise(() => {})
```

[playground](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbxAQxgYwBYF84DMoQhwDkMAzgLRiowCmUAdsQFDNoQNnwDCA8gLIAFXgDkAoiIAqAZQD6YgBqSJ0gJKjpcALxwA2sQB0ANy7EANCWNcAzOcvtwHWgxh3DyAK5uDyZAA8wABsPAHNgJgt3MDBiAF04ZDI4dk4YZjgMzOYAE1o0QOQoWmSOLjhaPzpOYA4ALjguKHCQ1hR0DAAKCqqyGoYASnS4AwB3YBhO4jAAaxC7RIBPBjQ4Dv7tAD5ELEGM0fHOgyO+IVEJGXklFXURaQtF5dW-da0thB2huAB6L8y--4BgKBmQAegB+IYGCATehjMi0DprTbbfpAA)

**Reply by syabro** [reply]
Got it. I'm generating array dynamically.

<img width="807" alt="image" src="https://github.com/gvergnaud/ts-pattern/assets/29029/897d0bb3-4acf-493e-bd7b-1d3cd1c25e6e">

[playground](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbxAQxgYwBYF84DMoQhwDkMAzgLRiowCmUAdsQFDNoQNnwDCA8gLIAFXgDkAoiIAqAZQD6AEQCSAJWlwAvImZwSAOgBuXYgC4SAegAywAEZRkUAJ5mAggFcAJsAhnBAG1cA5hSKnGYAatKSxAA02nqGMADMJuZWtvZObp7efoHBoRGSybE6xLrs4By0DDApxJY2do4uHl4+-kEhZGbchJAM1eQxcWXIrnUN6c1Zbbmdob2VAzVkw6W6yMgAHmAdwEym9WlNmWC7wGioXgxw0q5n0DAu+sDuLdnteV1reshnE84zr4LlcOKtYlhWLQdo84OxODwBMJxFI5GIABqSCTSRSiNSaXjWABWtDQMF0AGtaA4yAAKPhCUQSGQKFTSACUcR0OmY7lJvnstDhYPg0LonGupi4UH2AVYKHQGFpYuqZGunJ0ugA7sAYEriGAKQEYnBkGQHAw0HBaeyNAA+RBYDVwbW6pW6D0M5HMtGY7G4kTSaKm82W61bW3qB0IJ1csxmbmJpPJlOp7kAPQA-HFdBA9fQdWRaLSbfbHeygA)
